tests/console/teuthology.pm

```diff
@@ -0,0 +1,81 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run teuthology suites via ECP or OVH openstack and get results and logs
+# Maintainer: Jozef Pupava <jpupava@suse.com>
+
+use base "consoletest";
+use strict;
+use testapi;
+use version_utils 'is_sle';
+
+my $incidentnr  = get_var('INCIDENTNR');
+my $suite       = get_var('TEUTHOLOGY_SUITE');
+my $ses_version = is_sle('=15') ? 'ses6' : 'ses5';
+
+sub upload_teuthology_logs {
+    # get all teuthology logs, tar them and upload
+    upload_logs "teuthology-openstack-$suite.log";
+    assert_script_run "export COUNT=\$(grep openstack/[[:digit:]]/teuthology teuthology-openstack-$suite.log|wc -l)";
+    assert_script_run "for c in `seq 1 \$COUNT`; do for l in `grep \"openstack/\$c/teuthology.log\" \\
+teuthology-openstack-$suite.log|awk '{print\$6}'`;do wget \$l && mv teuthology.log teuthology_target_\$c.log; done; done";
+    assert_script_run "tar zcvf teuthology_targets.tar.gz teuthology_target_*";
+    upload_logs 'teuthology_targets.tar.gz';
+}
+
+sub run {
+    select_console 'root-console';
+    # export variables for teardown of openstack VM in next test
+    assert_script_run "export incidentnr=$incidentnr";
+    assert_script_run "export ses_version=$ses_version";
+    # TODO select ECP or OVH, now is ECP preconfigured in disk image
+    assert_script_run 'openstack keypair list';
+    assert_script_run 'openstack server list';
+    # parse maintenance repos into variable for teuthology repos
+    my @repos = split(/,/, get_var('MAINT_TEST_REPO'));
+    my $repo_count = 1;
+    my $maint_test_repos;
+    while (defined(my $maintrepo = shift @repos)) {
+        next if $maintrepo =~ /^\s*$/;
+        $maintrepo = "--test-repo maint_repo$repo_count:$maintrepo";
+        $maint_test_repos .= " " . $maintrepo;
+        $repo_count++;
+    }
+    # run teuthology openstack in ECP or OVH
+    assert_script_run "teuthology-openstack -v --key-filename ~/.ssh/id_rsa --key-name QAM-openqa \\
+--name QAM-$incidentnr-openqa-$ses_version --simultaneous-jobs 20 \\
+--teuthology-git-url http://github.com/SUSE/teuthology --teuthology-branch master \\
+--suite-repo http://github.com/SUSE/ceph --suite-branch $ses_version \\
+--ceph-repo http://github.com/SUSE/ceph --ceph $ses_version --suite $suite --filter sle \\
+--test-repo sle12_repo:http://download.suse.de/ibs/SUSE:/SLE-12-SP3:/GA/images/repo/SLE-12-SP3-Server-POOL-x86_64-Media1 \\
+--test-repo ses5_repo:http://download.suse.de/ibs/SUSE/Products/Storage/5/x86_64/product \\
+$maint_test_repos --wait |& tee teuthology-openstack-$suite.log; if [ \${PIPESTATUS[0]} -ne 0 ]; then false; fi", 7000;
```

Comment:
Maybe set `MAX_JOB_TIME` on test if some commands should take that long..

Reply:
Right now this should be more then enough, there are no suites that I know that will take longer
Test will pass or fail much sooner, I just choose 7000 to avoid timeout